### DISCUSSION TITLE
Produktyzacja - propozycja zmiany

### DISCUSSION THREAD

**Discussion by glorifyingpeace** [allegro/allegro-api#6372] [Suggestions - in progress]
Występuje problem: 
 - [x ] z zasobem REST API / with a REST API resource
 - [x ] ze środowiskiem produkcyjnym / with the production environment
 - [ ] ze środowiskiem testowym / with the test environment
 - [ ] z logowaniem lub autoryzacją / with login or authorization
 - [ ] na portalu https://developer.allegro.pl/ / at https://developer.allegro.pl/en/
 - [ ] inny / other

:beetle: **Opis / Description**

do tworzenia produktów używamy
https://api/.{environment}/sale/product-offers

Przykładowy response:
HTTP/2 422 
expires: 0
cache-control: no-cache, no-store, max-age=0, must-revalidate
x-xss-protection: 1; mode=block
pragma: no-cache
x-frame-options: DENY
trace-id: f28edca20c004697
date: Tue, 15 Nov 2022 08:44:45 GMT
x-content-type-options: nosniff
strict-transport-security: max-age=31536000 ; includeSubDomains
content-type: application/json
content-length: 512

{"errors":[{"code":"ProductValidationException","message":"Product has invalid values [ValidatedParameter{parameter=SimpleParameter{id='3786', values=[ParameterValue{id='3786_85', value='inna'}]}, name='Marka', validationErrors=[ValidationError{value='inna', valueId='3786_85', errorType=VALUE_AMBIGUOUS, illegalCharacters='null'}]}] in category: 76034","details":null,"path":"productSet[0].product","userMessage":"Nieprawidłowe wartości parametrów: [parameter = Marka [ Wartość niejednoznaczna - inna]]"}]} 

aby prawidłowo obsłużyć taki error w body responsa brakuje nam id produktu allegro, w związku z czym nie możemy wysłać propozycji zmian albo wyświetlić użytkownikowi jak wygląda ten produkt aby zdecydował czy na weno chce się łączyć z tym produktem.

Dlaczego tak?:
ponieważ niejednokrotnie allegro przez ten endpoint chce łączyć naszą ofertę z tym konkretnym produktem, ale robi to niewłaściwie ponieważ : ktoś inny użył przy produkcie zły gtin, mpn lub ean. W kategoriach związanych z odzieżą jest to nagminne. Dodatkowo zauważyliśmy że eany mpn odzieżowe niejednokrotnie są używane przy produktach z poza moda. W modzie nowe modele produktów są dodawane w rekordowym tępię często nie posiadają ean mpn lub gtin.

Rozwiązania opcje: 
 rezygnacja z produktyzacji w modzie w oparciu o mpn gtin ean jako wymagane,  stworzenie własnego numeru identyfikującego produkt, tak jak to jest na [amazon.com](x-webdoc://BCAB2E83-13C6-4820-9D11-E3BB2A4A449E/amazon.com) gdzie funkcjonuje ASIN
dodanie response  https://api/.{environment}/sale/product-offers , id produktu ponieważ jest wymagany w https://api/.{environment}/sale/products/{productId}/change-proposals

**Comment by MartaNowaczyk** [comment]
Dziękuję za Twoje zgłoszenie. Jest to jednak sugestia, dlatego przenoszę ją do zakładki "Discussions". Dzięki temu również inni użytkownicy będą mogli na nią zagłosować, co dodatkowo określi, jak bardzo priorytetowa jest to potrzeba.

**Comment by mbcont** [comment]
witam, uważam podobnie i podbijam temat

**Comment by MartaNowaczyk** [comment]
Po przeanalizowaniu tematu przez odpowiedni zespół, na ten moment proponowane rozwiązanie nie jest możliwe do wdrożenia, dlatego sugestię oznaczam jako odrzuconą. Jeśli w przyszłości pojawi się taka możliwość, by uzyskać taki lub podobny rezultat, poinformujemy o tym w oddzielnym komunikacie
